### DISCUSSION TITLE
Dynamic Collection Name?

### DISCUSSION THREAD

**Discussion by DizzyDuan** [isar/isar#445] [Help] [answered]
We need to create a separate collection for each chat session

Dynamic creation and operation based on session ID

Can someone help me?

I hope it can be used like this：

        final isar = await Isar.open(
          schemas: [QuoteSchema("user_123456")],
          directory: (await getApplicationDocumentsDirectory()).path,
        );
        isar.quotes("user_123456").where().limit(1).build().watch(initialReturn: true);

**Comment by simc** [comment, marked as answer]
Dynamic collection names are not supported and not a good idea.

Please add a `chatSession` field or something similar to your collection and index it for better performance.
